File: backend/app/api/config_platform.py
"""Configuration platform (CATIA) integration stubs — to be implemented after intranet deployment."""
from fastapi import APIRouter, Depends
from pydantic import BaseModel
from app.models.user import User
from app.api.deps import get_current_user

router = APIRouter(tags=["config-platform"])

class FetchModelsRequest(BaseModel):
    equipment_ids: list[str]

@router.get("/config-platform/status")
async def platform_status(user: User = Depends(get_current_user)):
    return {"connected": False, "message": "待内网部署后对接配置平台(CATIA)"}

@router.post("/config-platform/fetch-models")
async def fetch_models(body: FetchModelsRequest, user: User = Depends(get_current_user)):
    return {
        "status": "not_available",
        "message": "外网环境无法连接配置平台，请在内网环境使用此功能",
        "requested_count": len(body.equipment_ids),
    }

File: frontend/src/api/config-platform.ts
import client from './client';

export async function getPlatformStatus(): Promise<{ connected: boolean; message: string }> {
  const { data } = await client.get('/config-platform/status');
  return data;
}

export async function fetchModels(equipmentIds: string[]): Promise<{ status: string; message: string }> {
  const { data } = await client.post('/config-platform/fetch-models', { equipment_ids: equipmentIds });
  return data;
}
